File: java/examples/VaasExample/src/main/java/de/gdata/vaasexample/Authentication.java
package de.gdata.vaasexample;

import de.gdata.vaas.*;
import de.gdata.vaas.authentication.*;
import java.net.URI;


public class Authentication {
    public static void main(String[] args) throws Exception {
        var clientId = System.getenv("CLIENT_ID");
        var clientSecret = System.getenv("CLIENT_SECRET");
        var vaasclientId = System.getenv("VAAS_CLIENT_ID");
        var userName = System.getenv("VAAS_USER_NAME");
        var password = System.getenv("VAAS_PASSWORD");
        var tokenUrl = System.getenv("TOKEN_URL");
        if (tokenUrl == null) { tokenUrl = "https://account.gdata.de/realms/vaas-production/protocol/openid-connect/token"; }
        var vaasUrl = System.getenv("VAAS_URL");
        if (vaasUrl == null) { vaasUrl = "wss://gateway.production.vaas.gdatasecurity.de"; }

        // If you got a username and password from us, you can use the ResourceOwnerPasswordAuthenticator like this
        var authenticator = new ResourceOwnerPasswordGrantAuthenticator(
                vaasclientId,
                userName,
                password,
                new URI(tokenUrl)
        );
        // You may use self registration and create a new username and password for the
        // ResourceOwnerPasswordAuthenticator by yourself like the example above on https://vaas.gdata.de/login

        // Else if you got a client id and client secret from us, you can use the ClientCredentialsGrantAuthenticator like this
        // var authenticator = new ClientCredentialsGrantAuthenticator(
        //         clientId,
        //         clientSecret,
        //         new URI(tokenUrl)
        // );

        var config = new VaasConfig(new URI(vaasUrl));
        var vaas = new Vaas(config, authenticator);
        var verdict = vaas.forSha256(new Sha256("275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f"));
        System.out.printf("File %s was detected as %s", verdict.getSha256(), verdict.getVerdict());
    }
}

File: ruby/examples/authentication.rb
require 'vaas/client_credentials_grant_authenticator'
require 'vaas/resource_owner_password_grant_authenticator'


def main
    client_id = ENV.fetch("CLIENT_ID") || "vaas-customer"
    client_secret = ENV.fetch("CLIENT_SECRET")
    user_name = ENV.fetch("VAAS_USER_NAME")
    password = ENV.fetch("VAAS_PASSWORD")
    token_url = ENV.fetch("TOKEN_URL") || "https://account.gdata.de/realms/vaas-production/protocol/openid-connect/token"
    vaas_url = ENV.fetch("VAAS_URL") || "wss://gateway.production.vaas.gdatasecurity.de"
    test_url = "https://gdata.de"

    #If you got a username and password from us, you can use the ResourceOwnerPasswordAuthenticator like this
    authenticator = VAAS::ResourceOwnerPasswordGrantAuthenticator.new(
      client_id,
      user_name,
      password,
      token_url
    )
    # You may use self registration and create a new username and password for the
    # ResourceOwnerPasswordAuthenticator by yourself like the example above on https:#vaas.gdata.de/login

    # Else if you got a client id and client secret from us, you can use the ClientCredentialsGrantAuthenticator like this
    # authenticator = VAAS::ClientCredentialsGrantAuthenticator.new(
    #   client_id,
    #   client_secret,
    #   token_url
    # )


    vaas = VAAS::VaasMain.new(vaas_url)
    token = authenticator.get_token

    Async do
        vaas.connect(token)

        verdict = vaas.for_url(test_url)
        puts "Verdict #{verdict.wait.sha256} is detected as #{verdict.wait.verdict}"

        vaas.close
    end
end


if __FILE__  == $0
    main
end